### PR TITLE
Adds OpenAPI parsing as a gate

### PR DESCRIPTION
**Pull request by zengin**
- partial https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/issues/782

### CODE REVIEW

**Comment by zengin** [comment]
Test workflow run with broken OpenAPI doc: https://github.com/microsoftgraph/msgraph-metadata/actions/runs/2573308096

**Review by andrueastman — APPROVED** [review]

File: tools/OpenAPIParser/OpenAPIParser.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
```
Comment (baywet) [inline tools/OpenAPIParser/OpenAPIParser.csproj:3]: Why not use hidi with the validate command instead?

Comment (zengin) [reply tools/OpenAPIParser/OpenAPIParser.csproj:3]: This (what kiota and DevX API do) was the quickest I could find at 7pm while dealing with the ICM and the time pressure to block today's generation with a gate :) I opened an issue to address this referencing this comment.

Comment (baywet) [reply tools/OpenAPIParser/OpenAPIParser.csproj:3]: thanks for dealing with this! I do understand the urgency aspect of this PR.

**Pull request merged**